pkg/build/api/v1/types.go

```diff
@@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
 
 	// Git is the git information if the Type is BuildSourceGit
 	Git *GitInfo `json:"git,omitempty"`
+
+	// Env contains additional environment variables you want to pass into a builder container
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
> @liggitt - were you referring to something other than an on/off switch?

Not sure. Possibly a whitelist of allowed envvar names. I don't want to make it too hard to use, but I also don't want the default webhook config to let the remote site do overly powerful things. I don't really know what is possible via envvar, but knowing that git lets you ignore TLS errors and curl lets you redirect requests through a proxy via envvar makes me wary.